src/main/scala/piconot/external/Parser.scala

```diff
@@ -0,0 +1,68 @@
+package piconot.external
+
+import scala.util.parsing.combinator._
+
+object PicobotParser extends JavaTokenParsers with PackratParsers {
+    def apply(s: String): ParseResult[Program] = parseAll(program, s)
+    
+    def program: Parser[Program] = {
+        "start" ~ name ~ repsep(execution, "") ^^ {case "start" ~ n ~ es ⇒ Program(n, es)}
+    }
+
+    def execution: Parser[Execution] = {
+        ( task ^^ {t ⇒  t}
+         | rule  ^^ { r ⇒ r})
+    }
+    
+    def task: Parser[Task] = {
+        "task" ~ name ~ ":" ~ "start" ~ name ~ repsep(rule, "") ^^ {case "task" ~ n ~ ":" ~ "start" ~ s ~ rs ⇒ Task(n, s, rs)}
```

Comment:
Some of these lines end up getting pretty long, though I suppose that's a reflection of a wordier syntax and descriptive variable names. I also don't think you need to use `repsep` here, Scala should automatically deal with whitespace.